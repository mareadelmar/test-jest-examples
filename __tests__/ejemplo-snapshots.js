const cliente = {
	nombre: "Juan",
	balance: 500,
	tipo: "premium",
};

// en el ejemplo de objetos el test era sobre las propiedades, no sobre todo el objeto.
// para eso podríamos convertirlo en string, por ej.
// Pero además jest ofrece algo llamado snapshots: datos que se almacenan en un strings, en una carpeta aparte, sobre la que podemos comparar los datos.

describe("Cliente", () => {
	test("test a todo el objeto", () => {
		expect(cliente).toMatchSnapshot(); // cuando la ejecutemos se creará la carpeta snapshot
	});
});

/*
Si ahora a este objeto cliente de arriba le cambiamos algún valor, la prueba no pasará
y jest nos mostrará las diferencias entre este y el objeto guardado en snapshot.
A menos que ejecutemos el test con la flag -- -u (o sea, npm test -- -u): esto actualizará el snapshot con los nuevos valores.
(Otra opción si los valores del snapshot ya no nos sirven es eliminarlo y que se vuelva a crear; 
mejor nunca ir a modificar el archivo hardcodeando).

*/
