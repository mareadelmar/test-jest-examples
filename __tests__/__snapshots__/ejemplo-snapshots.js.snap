// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cliente test a todo el objeto 1`] = `
Object {
  "balance": 500,
  "nombre": "Juan",
  "tipo": "premium",
}
`;
